internal/builds/builds.go

```diff
@@ -0,0 +1,97 @@
+package builds
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/broadinstitute/sherlock/internal/services"
+	"gorm.io/gorm"
+)
+
+// BuildController is the management layer that processes requests
+// to the /builds api group
+type BuildController struct {
+	store buildStore
+	// this is needed so that we can automatically create a new service entity
+	// if a build is reported for a service not tracked by sherlock
+	services *services.ServiceController
+}
+
+// NewController returns an instance of the controller struct for
+// interacting with build entities. It embeds a buildStore interface for
+// operations on the build persistence layer
+func NewController(dbConn *gorm.DB) *BuildController {
+	buildStore := newBuildStore(dbConn)
+	return &BuildController{
+		store:    buildStore,
+		services: services.NewController(dbConn),
+	}
+}
+
+// Response is a type that allows all data returned from the /builds api group to share a consistent structure
+type Response struct {
+	Builds []Build `json:"builds"`
+	Error  string  `json:"error,omitempty"`
+}
+
+// CreateBuildRequest is the type used to validate a request for a new build
```

Comment:
Should there be a separate struct type here for serializing Builds in responses? I realize it adds overhead but it's a bit surprising that fields in requests are snake case and fields in responses are camel case. And I think it will be useful later on (for hiding model details in the API, maintaining backwards compatibility, populating some fields w/ code, etc).

Reply:
great point. I think I will tackle this in a separate ticket though!

Reply:
[DDO-1532](https://broadworkbench.atlassian.net/browse/DDO-1532)